### PR TITLE
kdb: kdbCommit

### PR DESCRIPTION
**Pull request by vLesk**
Implementing a new plugin function, as discussed in #2798.

## Basics

Check relevant points but **please do not remove entries**.
Do not describe the purpose of this PR in the PR description but:

- [x] Short descriptions should be in the release notes (added as entry in
      `doc/news/_preparation_next_release.md` which contains `_(my name)_`)
      **Please always add something to the the release notes.**
- [x] Longer descriptions should be in documentation or in design decisions.
- [x] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests
- [x] I ran all tests locally and everything went fine
- [ ] affected documentation is fixed
- [ ] I added code comments, logging, and assertions (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] meta data is updated (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))

## Review

Reviewers will usually check the following:

- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)
- [ ] Documentation is concise and good to read
- [ ] Examples are well chosen and understandable

## Labels

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the build server is happy** and also you
  say that everything is ready to be merged.

### CODE REVIEW

File: doc/decisions/README.md [hunk 1]
```diff
@@ -50,3 +50,4 @@ section here.
 - [Default Values](default_values.md)
 - [High Level API](high_level_api.md)
 - [Global KeySet](global_keyset.md)
+- [Commit Function](commit_function.md)
```
Comment (markus2330) [inline doc/decisions/README.md]: You can move it up to decided.

File: src/plugins/template/template.c [hunk 2]
```diff
@@ -69,6 +69,11 @@ int elektraTemplateError (Plugin * handle ELEKTRA_UNUSED, KeySet * returned ELEK
 	return ELEKTRA_PLUGIN_STATUS_SUCCESS;
 }
 
+int elektraTemplateCommit (Plugin * handle ELEKTRA_UNUSED, KeySet * returned ELEKTRA_UNUSED, Key * parentKey ELEKTRA_UNUSED)
+{
+	return elektraTemplateSet (handle, returned, parentKey);
```
Comment (markus2330) [inline src/plugins/template/template.c]: We should not recommend to do this. It is better to keep set and commit logic separated. (Only for migration purposes it is fine.)

**Comment by markus2330** [comment]
Ahh good to see commits from you :+1:

**Comment by vLesk** [comment]
Thank you! :) Also, thank you for the comments! I am currently struggling with suppressing the offending symbol correctly. According to the server, the ELEKTRA_PLUGIN_COMMIT enum I added into src/include/kdbplugin.h should be the problem. I have tried entering that, as well as plugin_t (the enum) into icheck.suppression but still get the same error from the server. Am I entering the enum wrong or is there something else I should enter?

**Comment by markus2330** [comment]
The suppression is quite primitive, it simply cuts out lines containing the words from tests/icheck.suppression but maybe scripts/run_icheck has a bug?

ELEKTRA_PLUGIN_COMMIT should do the magic. But I can reproduce that it does not work.
If I add it, I get the log output:
Remove entry matching “ELEKTRA_PLUGIN_COMMIT” from new interface
but also the same error as you get.

@sanssecours do you have any idea?

**Comment by markus2330** [comment]
Ohh, I see that in d521abf1e96ed3942380d1d089fc13b3c4b1a2db @KurtMi changed ex with ed.

I would have written in sed, as I usually do not use ex nor ed.

**Comment by sanssecours** [comment]
> @sanssecours do you have any idea?

I added the entry `ELEKTRA_PLUGIN_COMMIT` to [`icheck.suppression`](https://github.com/ElektraInitiative/libelektra/blob/master/tests/icheck.suppression). As far as I can tell this seems to fix the problem, at least on my machine. I pushed a commit that contains the fix here too.

> Am I entering the enum wrong or is there something else I should enter?

I think the problem might be that you did not add a line feed (`\n`) before the end of the file. Normally this should not be a problem, but [`run_icheck`](https://github.com/ElektraInitiative/libelektra/blob/master/scripts/run_icheck) is a rather primitive script that might not take missing newlines into account.

**Comment by markus2330** [comment]
Thank you for finding the problem! We should add this as comment (that a new-line is needed at the end of the file).

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by vLesk** [comment]
Thank you very much for your help! The suppression works fine now.

**Comment by vLesk** [comment]
The server seems to be satisfied with my implementation, @markus2330 please review my PR.

**Comment by markus2330** [comment]
Thank you, great work!

There is one little leftover, I described it in  #2918. You can fix this together with the next PR (or also create a separate PR for it, as you like).

**Comment by vLesk** [comment]
Thank you! I'll take care of the issue.

**Pull request merged**